nav2_costmap_2d/src/costmap_subscriber.cpp

```diff
@@ -26,72 +27,130 @@ CostmapSubscriber::CostmapSubscriber(
 : topic_name_(topic_name)
 {
   auto node = parent.lock();
+  logger_ = node->get_logger();
   costmap_sub_ = node->create_subscription<nav2_msgs::msg::Costmap>(
     topic_name_,
     rclcpp::QoS(rclcpp::KeepLast(1)).transient_local().reliable(),
     std::bind(&CostmapSubscriber::costmapCallback, this, std::placeholders::_1));
+  costmap_update_sub_ = node->create_subscription<nav2_msgs::msg::CostmapUpdate>(
+    topic_name_ + "_updates",
+    rclcpp::QoS(rclcpp::KeepLast(10)).transient_local().reliable(),
```

Comment:
I was just wondering if you could kindly explain why a queue size of '10' was chosen?

Reply:
Updates are generated only for the parts of the map where the change has occurred (new cleaning or marking observation). It might happen that some changes will be available only in a single update msg. So I guess that my intuition was to extend the queue size to prevent losing this information when the subscriber is waiting for the collision checker or other clients to end their computations. 
Size 10 was just arbitrarily selected as a default value for QoS in ROS 2. I am open to suggestions and discussion if the queue is even necessary here.